document GHC-55666 (#427)
haskellfoundation/error-message-index: message-index/messages/GHC-55666/index.md

---
title: Strictness annotation on unlifted type
summary: Using a strictness annotation (bang) on an unlifted type is redudant as unlifted values are strict by definition
severity: warning
flag: -Wredundant-strictness-flags
introduced: GHC 9.6.1
---

> A strictness annotation (also called a bang: `!`) can be used to denote that a type should not be evaluated lazily.

Reviewer: This could lead to some confusion, because there's not such thing as "evaluating a type". Types are actually erased during compilation and only description of value-level computation (my amateurish formulation) remain.

I suggest reformulating this to something like ".. can be used to mark fields within data types, that should not be evaluated lazily".

In some cases this can lead to faster code because fewer heap allocation are required.

However, unlifted types like `Int#` are strict by definition because they are a value,
not a pointer to a potentially unevaluated value (thunk).

> Therefore, adding strictness annotations to unlifted types or fields of such types are redudant.

Reviewer: ```suggestion
Therefore, adding strictness annotations to unlifted types or fields of such types is redundant.
```

---
title: Strictness annotation on unlifted type
summary: Using a strictness annotation (bang) on an unlifted type is redudant as unlifted values are strict by definition
severity: warning
flag: -Wredundant-strictness-flags
introduced: GHC 9.6.1
---

A strictness annotation (also called a bang: `!`) can be used to denote that a value should not be evaluated lazily.

> In some cases this can lead to faster code because fewer heap allocation are required.

Reviewer 2: ```suggestion
In some cases this can lead to faster code because fewer heap allocations are required.
```

Here it is used to mark a that a field in a datatype should not be evaluated lazily.

However, unlifted types like `Int#` are strict by definition because they are a value,
not a pointer to a potentially unevaluated value (thunk).

> Therefore, adding strictness annotations to unlifted types or fields of such types are redundant.

Reviewer 2: ```suggestion
Therefore, adding strictness annotations to unlifted types or fields of such types is redundant.
```
